library/psa_crypto_slot_management.c

```diff
@@ -257,6 +272,36 @@ psa_status_t psa_get_key_slot( mbedtls_svc_key_id_t key,
 
 }
 
+psa_status_t psa_decrement_key_slot_access_count( psa_key_slot_t *slot )
+{
+    if( slot == NULL )
+        return( PSA_SUCCESS );
+
+    if( slot->access_count > 0 )
+    {
+        slot->access_count--;
+        return( PSA_SUCCESS );
+    }
+
+    /*
+     * As the return error code may not be handled in case of multiple errors,
+     * do our best to report if the access counter is equal to zero: if
+     * available call MBEDTLS_PARAM_FAILED that may terminate execution (if
+     * called as part of the execution of a unit test suite this will stop the
+     * test suite execution) and if MBEDTLS_PARAM_FAILED does not terminate
+     * execution ouput an error message on standard error output.
+     */
+#ifdef MBEDTLS_CHECK_PARAMS
+    MBEDTLS_PARAM_FAILED( slot->access_count > 0 );
+#endif
+#ifdef MBEDTLS_PLATFORM_C
+    mbedtls_fprintf( stderr,
```

Comment:
It seems that if MBEDTLS_PLATFORM_C is defined then mbedtls_fprintf() is defined. It may not do anything but it is define thus having this seems acceptable to me. As I explain in the comment above, the idea is to maximize our chance to spot when something goes wrong. In the comment I should probably say "try to output an error message on standard error output if possible" rather than just "output an error message on standard error output" though.
Regarding consistency the case of `psa_xxx_abort` functions is maybe a bit different as we don't expect their returned code to be ignored in some cases while we expect it for `psa_decrement_key_slot_access_count()`.

Reply:
But outputting to stderr might not be right. For example, this might be a log file which must be in a certain format, or a log file that's visible to an attacker that we don't want to give information to. Adding stderr feedback is a functional and security design change.

Reply:
I am not completely convinced but I removed it as this discussion goes significantly beyond the scope of this PR I'd say.